### ISSUE TITLE
Provide a way to only lint certain file types

### ISSUE THREAD

**Issue by andrewnicols** [DavidAnson/markdownlint-cli2#45]
Maybe I'm just missing something, but if I want to lint all markdown files in a directory, there does not seem to be any way to do so easily.

If I specify a glob in the config file, this only works for the current dir.
```
globs:
  - '*.md'
```

If I have it check subdirs, then I have no way of linting only a single file any more:
```
globs:
  - **/*.md
```

If I specify an ignores, then this works:
```
ignores:
  - '**/*.png'
  - '**/*.js'
```

But requires that I know every possible extension that could be used in my project (not practical).

I have tried a negated ignore:
```
ignore:
  - '!*.md'
```

But that becomes `!!*.md` which is apparently not valid.

Basically, I want to be able to search all markdown files as standard, but if I specify a specific path I want only that path to be tested, and I want to be able to check a whole directory or set of directories (which may include unrelated files).

It would be nice if there were two additional options:

 - `defaultGlobs` - the default glob to use if one was not specified on the CLI
 - `extensions` - a list of file extensions applied as standard

**Comment by DavidAnson** [comment]
I think the root of your difficulty is that you are trying to put the list of globs in the config file. That list is typically passed on the command line which supports multiple glob expressions (including ignores) which can be used to customize each run to target exactly the files you want. The scenarios you describe (single file, current directory, all files, exclusions) should all be expressable in this way.

If this explanation doesn't help, please give a specific example of a directory tree and files you wish to lint and I will reply with a command that should do that.

**Comment by DavidAnson** [comment]
Your last paragraph seems to add a considerable amount of complexity to the tool in order to accomplish something that seems to me to be fairly straightforward via scripts. Each of your examples could be a script in npm which means you would need to educate authors that the script exist, but also saves them from needing to understand all the subtle rules about how to target specific scenarios. I don't know the size of your project, but I would love for the default answer to just be to lint everything you care about every time. I've tried to make the tool as fast as possible and for the most part expect it to run in less than a second. I test on some large repositories where it takes longer, and maybe that describes you? If so, I'd be happy to do a performance profile as there may be some thing dumb on my side that can be improved. I do a performance pass about once a year and I'm due for one soon, though I don't expect it to happen for the upcoming release.

**Comment by DavidAnson** [comment]
I did a performance pass on the library. I hope it will be fast enough for linting everything to be/become acceptable; that’s what I do for my projects. Closing this issue per my comments above.
